src/librbd/ImageState.h

```diff
@@ -73,6 +74,7 @@ class ImageState {
     ActionType action_type;
     uint64_t refresh_seq = 0;
     std::string snap_name;
+    cls::rbd::SnapshotNamespace snap_namespace;
```

Comment:
You need to update ```ImageState::set_snap``` to take a snapshot namespace and assign the snapshot namespace to this param.